src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx

```diff
@@ -172,6 +173,8 @@ function ReportActionCompose({
      * Shows red borders and prevents the comment from being sent
      */
     const {hasExceededMaxCommentLength, validateCommentMaxLength} = useHandleExceedMaxCommentLength();
+    const {hasExceededMaxTitleLength, validateTitleMaxLength} = useHandleExceedMaxTaskTitleLength();
```

Comment:
sure

Reply:
> I don't think it's a good idea to check conditions based on result of a debounce function. It's better if we detect comment format and use properly validation function here

In my opinion using a separate variable like `isCreatingTaskComment` would be redundant as we already perform regex matching in `useHandleExceedMaxTaskTitleLength`. If we implement leading true, it should be safe. But let me know if we still need to do that.

Reply:
@hoangzinh after test it might looks like this

src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx
```tsx

    const onValueChange = useCallback(
        (value: string) => {
            if (value.length === 0 && isComposerFullSize) {
                Report.setIsComposerFullSize(reportID, false);
            }

            const taskCommentMatch = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
            if (taskCommentMatch) {
                const title = taskCommentMatch?.[3] ? taskCommentMatch[3].trim().replace(/\n/g, ' ') : '';
                validateTaskTitleMaxLength(title);
            } else {
                validateCommentMaxLength(value, {reportID});
            }
        },
        [isComposerFullSize, reportID, validateCommentMaxLength, validateTaskTitleMaxLength],
    );


```


src/hooks/useHandleExceedMaxTaskTitleLength.ts
```tsx

const useHandleExceedMaxTaskTitleLength = () => {
    const [hasExceededMaxTaskTitleLength, setHasExceededMaxTitleLength] = useState(false);

    const handleValueChange = useCallback((title: string) => {
        const exceeded = title ? title.length > CONST.TITLE_CHARACTER_LIMIT : false;
        setHasExceededMaxTitleLength(exceeded);
    }, []);

    const validateTaskTitleMaxLength = useMemo(() => debounce(handleValueChange, CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME, {leading: true}), [handleValueChange]);

    return {hasExceededMaxTaskTitleLength, validateTaskTitleMaxLength};
};


```

proceed with the above change? or alternatively we can keep current `useHandleExceedMaxTaskTitleLength` logic but no need to debounce it?

Reply:
@hoangzinh will 100ms works? because each function has inner debounce value

`validateTaskTitleMaxLength` (500+100)  `validateCommentMaxLength` (1500+100)

Reply:
@hoangzinh if we're going to extract the debounce and not need faster debounce specific for task title, how about keeping the regex matching inside the hook?

```tsx

const useHandleExceedMaxTaskTitleLength = () => {
    const [hasExceededMaxTaskTitleLength, setHasExceededMaxTitleLength] = useState(false);

    const validateTaskTitleMaxLength = useCallback((value: string) => {
        const match = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
        if (match) {
            const title = match[3] ? match[3].trim().replace(/\n/g, ' ') : undefined;
            const exceeded = title ? title.length > CONST.TITLE_CHARACTER_LIMIT : false;
            setHasExceededMaxTitleLength(exceeded);
            return true;
        }
        setHasExceededMaxTitleLength(false);
        return false;
    }, []);

    return {hasExceededMaxTaskTitleLength, validateTaskTitleMaxLength};
};


```

Reply:
@hoangzinh the code has been updated. I use and modify `COMMENT_LENGTH_DEBOUNCE_TIME` since it is not being used anywhere.